core/src/App.svelte

```diff
@@ -942,7 +945,7 @@
     const showModalPathInUrl = LuigiConfig.getConfigBooleanValue(
       'routing.showModalPathInUrl'
     );
-    if (showModalPathInUrl) {
+    if (showModalPathInUrl && mfModalList.length===1) {
```

Comment:
Can you add a comment here as to why you selected 1 specifically. It's a very specific reason and might be lost over time.